### ISSUE TITLE
Problems integrating yaml-cpp into project. Not using find_package?

### ISSUE THREAD

**Issue by acampove** [jbeder/yaml-cpp#1330]
Hi,

I am reading [this](https://github.com/jbeder/yaml-cpp?tab=readme-ov-file#how-to-integrate-it-within-your-project-using-cmake) after an hour wasted trying to build my project against yours. I normally do:

```cmake
find_package(YAML-CPP REQUIRED yaml-cpp)

if (NOT YAML-CPP_FOUND)
...
```

Which works with all the C++ projects I have ever used in the last 15 years. But you do not seem to do things this way. For some reason you seem to require the user to use `FETCHCONTENT`. It would be very beneficial for 99% of the people out there, who do not know what `FETCHCONTENT` is that you update the documentation.

Cheers.

**Comment by jbeder** [comment]
I'm not sure I understand. Is the documentation you linked to incorrect? If so, happy to update it. Also, happy to update the CMake file if you have a suggestion. I'm not a CMake expert at all, and basically just let people boss me around for whatever's in it.

**Comment by ahogen** [comment]
@acampove you did not indicate whether you are trying to integrate yaml-cpp as a system package dependency or a source/embedded (e.g. git submodule) dependency. You likely know that `find_package()` is looking for either a `Find<package>.cmake` or `<package>-config.cmake` file. This project creates the latter from a [template](https://github.com/jbeder/yaml-cpp/blob/master/yaml-cpp-config.cmake.in) at build time for use when the project is installed or packaged ([reference](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html)). If you are trying to find a system package and that is failing, consider inspecting and possibly appending to [`CMAKE_MODULE_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_MODULE_PATH.html).

@jbeder the pattern you have documented is perfectly fine when using the repo as a source dependency. This pattern is commonly documented by other large projects:

- CMake's own documentation: https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html#fetchcontent-and-find-package-integration
- Googletest https://google.github.io/googletest/quickstart-cmake.html
- Catch2 https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md
- fmt https://fmt.dev/11.0/get-started/

In short, imo, what you have now is fine. Maybe could be improved to show how to use `find_package` when installed from a DEB/RPM package, but I wouldn't remove the existing text.

**Comment by acampove** [comment]
> I'm not sure I understand. Is the documentation you linked to incorrect? If so, happy to update it. Also, happy to update the CMake file if you have a suggestion. I'm not a CMake expert at all, and basically just let people boss me around for whatever's in it.

Hi,

The documentation should contain:

_In order to integrate this project with CMAKE, add the following lines to the CMakeLists.txt_

```cmake
    find_package(YAML-CPP REQUIRED yaml-cpp)
    if (NOT YAML-CPP_FOUND)
        message(FATAL_ERROR "YAML-CPP found ${YAML-CPP_FOUND}")
    endif()
```

additionally, your cmake needs to define `YAML_CPP_VERSION`. TLDR, you need a change in the cmake config and another in the documentation.

**Comment by jbeder** [comment]
@ahogen Does the above text look reasonable? Based on e.g. fmt's documentation you linked to, this would be the "Installed" path? Would it be reasonable to essentially duplicate their three choices?
